src/main/java/meditracker/medication/MedicationManager.java

```diff
@@ -85,6 +85,97 @@ public static Medication getMedication(String name) throws MedicationNotFoundExc
         throw new MedicationNotFoundException();
     }
 
+    /**
+     * Gets the Medication object from the medications list.
+     * Uses the Medication quantity to retrieve medications from the list.
+     * Shows all medications that is equal to or less than specified quantity,
+     *
+     * @param quantity Quantity of the medication to retrieve in double
+     * @throws MedicationNotFoundException No Medication matching specified name found
+     */
+    public static void showMedicationsByQuantity(Double quantity) throws MedicationNotFoundException {
+        int medicationsFound = 0;
+
+        for (Medication medication : medications) {
+            Double medicationListQuantity = medication.getQuantity();
+            if (Double.compare(medicationListQuantity, quantity) < 0
+                    || Double.compare(medicationListQuantity, quantity) == 0) {
+                medicationsFound++;
+                Ui.printSpecificMed(medication);
+            }
+        }
+        if (medicationsFound == 0) {
+            throw new MedicationNotFoundException();
+        }
+    }
+
+    /**
+     * Gets the Medication object from the medications list.
+     * Uses the Medication name to retrieve medications from the list.
+     * Shows all medications that contains the name.
+     *
+     * @param name Name of the medication to find and retrieve
+     * @throws MedicationNotFoundException No Medication matching specified name found
+     */
+    public static void showMedicationsByName(String name) throws MedicationNotFoundException {
+        int medicationsFound = 0;
+        for (Medication medication : medications) {
+            if (medication.getName().toLowerCase().contains(name)) {
+                medicationsFound++;
+                Ui.printSpecificMed(medication);
+            }
+        }
+        if (medicationsFound == 0) {
+            throw new MedicationNotFoundException();
+        }
+    }
+
+    /**
+     * Gets the Medication object from the medications list.
+     * Uses the Medication expiry to retrieve medications from the list.
+     * Shows all medications that will expire by the year that the user has input.
+     *
+     * @param expiry Expiry of the medication to find and retrieve
+     * @throws MedicationNotFoundException No Medication matching specified name found
+     */
+    public static void showMedicationsByExpiry(String expiry) throws MedicationNotFoundException {
```

Comment:
Just a note that possible refactoring will be required if we change the data type of expiry date to `LocalDate` within `Medication` in the future

Reply:
Got it. For now I will leave it be.